collect_app/src/test/java/org/odk/collect/android/entities/EntitiesRepositoryTest.kt

```diff
@@ -757,5 +757,14 @@ abstract class EntitiesRepositoryTest {
         savedEntities = repository.getEntities("things")
         assertThat(savedEntities[0].properties.size, equalTo(1))
         assertThat(savedEntities[0].properties[0].first, equalTo("prop"))
+
+        /**
+         * Attempt to save again to ensure that duplicate properties are correctly compared,
```

Comment:
I think we can break this into two test now as it feels like we're accounting for two distinct (but very related cases):

1. When an entity is created with duplicate case-insensitive properties
2. When an entity is created with a new property that's a case-insensitive duplicate of an existing one. 

So basically we can have one test that just checks saving something like:

```
val entity = Entity.New(
    "1",
    "One",
    properties = listOf(Pair("prop", "value"), Pair("Prop", "value"))
)
```

And another that checks saving a new entity to an existing list with a case-insensitive duplicate property like:

```
val repository = buildSubject()
val entity = Entity.New(
    "1",
    "One",
    properties = listOf(Pair("prop", "value"))
)

repository.save("things", entity)
var savedEntities = repository.getEntities("things")
assertThat(savedEntities[0].properties.size, equalTo(1))
assertThat(savedEntities[0].properties[0].first, equalTo("prop"))

repository.save("things", entity.copy(properties = listOf(Pair("Prop", "value"))))
savedEntities = repository.getEntities("things")
assertThat(savedEntities[0].properties.size, equalTo(1))
assertThat(savedEntities[0].properties[0].first, equalTo("prop"))
```

I'm pretty sure that'll drive out both the `distintBy` and the `filterNot` processing we have to do and give us different fails if one or the other is missing. Mainly though, I think it'll make it easier to understand the intended behaviour from the tests.